osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs

```diff
@@ -392,8 +401,20 @@ protected override void LoadComplete()
                     {
                         Schedule(() =>
                         {
-                            rankApplauseSound.VolumeTo(applause_volume);
-                            rankApplauseSound.Play();
+                            if (skins.CurrentSkin.Value is LegacySkin)
+                            {
+                                // only play legacy "applause" sound if score rank is B or higher.
+                                if (score.Rank >= ScoreRank.B)
+                                {
+                                    rankLegacyApplauseSound.VolumeTo(applause_volume);
+                                    rankLegacyApplauseSound.Play();
+                                }
+                            }
+                            else
+                            {
+                                rankApplauseSound.VolumeTo(applause_volume);
+                                rankApplauseSound.Play();
+                            }
```

Comment:
Ehhhhhh... I don't like this. The fact that we have all these skin fallback mechanisms and yet resort to resolving `SkinManager` feels like a failure.

I'd honestly change this to do something like

```diff
diff --git a/osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs b/osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs
index e5ba9500ee..f227d01bee 100644
--- a/osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs
+++ b/osu.Game/Screens/Ranking/Expanded/Accuracy/AccuracyCircle.cs
@@ -4,6 +4,7 @@
 #nullable disable
 
 using System;
+using System.Collections.Generic;
 using System.Linq;
 using osu.Framework.Allocation;
 using osu.Framework.Audio;
@@ -85,9 +86,6 @@ public partial class AccuracyCircle : CompositeDrawable
         /// </summary>
         public static readonly Easing ACCURACY_TRANSFORM_EASING = Easing.OutPow10;
 
-        [Resolved]
-        private SkinManager skins { get; set; }
-
         private readonly ScoreInfo score;
 
         private CircularProgress accuracyCircle;
@@ -101,7 +99,6 @@ public partial class AccuracyCircle : CompositeDrawable
         private PoolableSkinnableSample swooshUpSound;
         private PoolableSkinnableSample rankImpactSound;
         private PoolableSkinnableSample rankApplauseSound;
-        private PoolableSkinnableSample rankLegacyApplauseSound;
 
         private readonly Bindable<double> tickPlaybackRate = new Bindable<double>();
 
@@ -263,11 +260,14 @@ private void load()
                 if (isFailedSDueToMisses)
                     AddInternal(failedSRankText = new RankText(ScoreRank.S));
 
+                var applauseSamples = new List<string> { applauseSampleName };
+                if (score.Rank >= ScoreRank.B)
+                    applauseSamples.Insert(0, @"applause");
+
                 AddRangeInternal(new Drawable[]
                 {
                     rankImpactSound = new PoolableSkinnableSample(new SampleInfo(impactSampleName)),
-                    rankApplauseSound = new PoolableSkinnableSample(new SampleInfo(applauseSampleName)),
-                    rankLegacyApplauseSound = new PoolableSkinnableSample(new SampleInfo("applause")),
+                    rankApplauseSound = new PoolableSkinnableSample(new SampleInfo(applauseSamples.ToArray())),
                     scoreTickSound = new PoolableSkinnableSample(new SampleInfo(@"Results/score-tick")),
                     badgeTickSound = new PoolableSkinnableSample(new SampleInfo(@"Results/badge-dink")),
                     badgeMaxSound = new PoolableSkinnableSample(new SampleInfo(@"Results/badge-dink-max")),
@@ -401,20 +401,8 @@ protected override void LoadComplete()
                     {
                         Schedule(() =>
                         {
-                            if (skins.CurrentSkin.Value is LegacySkin)
-                            {
-                                // only play legacy "applause" sound if score rank is B or higher.
-                                if (score.Rank >= ScoreRank.B)
-                                {
-                                    rankLegacyApplauseSound.VolumeTo(applause_volume);
-                                    rankLegacyApplauseSound.Play();
-                                }
-                            }
-                            else
-                            {
-                                rankApplauseSound.VolumeTo(applause_volume);
-                                rankApplauseSound.Play();
-                            }
+                            rankApplauseSound.VolumeTo(applause_volume);
+                            rankApplauseSound.Play();
                         });
                     }
                 }

```

And yes I am well aware that this changes behaviour (lazer samples will play rather than nothing below B rank, but I also think that's probably fine).